src/azul/transformer.py

```diff
@@ -162,7 +162,8 @@ def translate_field(cls, value: AnyJSON, field_type: FieldType, forward: bool =
     def translate_fields(cls,
                          doc: AnyJSON,
                          field_types: Union[FieldType, FieldTypes],
-                         forward: bool = True) -> AnyMutableJSON:
+                         forward: bool = True,
+                         path: list = None) -> AnyMutableJSON:
```

Comment:
Always use the generic types from `typing` for type annotation if possible.

Reply:
Note that tuples are usually cheaper than lists, especially when they are passed by value (a new copy of the value is made for each recursion level).

Reply:
Also, tuples are immutable so you can use the empty tuple as the default value and …

(I feel like I've already had this conversation.)